core/src/main/scala/cromwell/core/io/AsyncIo.scala

```diff
@@ -0,0 +1,78 @@
+package cromwell.core.io
+
+import akka.actor.{Actor, ActorLogging, ActorRef}
+import cromwell.core.path.BetterFileMethods.OpenOptions
+import cromwell.core.path.Path
+
+import scala.concurrent.duration._
+import scala.concurrent.{Future, Promise}
+import scala.language.postfixOps
+
+
+trait AsyncIo extends IoClientHelper { this: Actor with ActorLogging with IoCommandBuilder =>
+  
+  protected val ioTimeout = 3 minutes
+  
+  override private [core] def ioResponseReceive: Receive = {
+    case (promise: Promise[_], ack: IoAck[Any] @unchecked) =>
+      cancelTimeout(promise -> ack.command)
+      // This is not typesafe. 
+      // However the sendIoCommand method ensures that the command and the promise have the same generic type
+      // Which means as long as only the sendIoCommand method is used to send requests, and the ioActor honors his contract
+      // and send back the right context with the right response, the types are virtually guaranteed to match.
+      promise.asInstanceOf[Promise[Any]].complete(ack.toTry)
+      ()
+  }
+  
+  /**
+    * IMPORTANT: This loads the entire content of the file into memory !
```

Comment:
Does this get called for things like `read_string()`, if so this is more a warning for wdl authors, right?

This might tie to #1762